packages/strapi-admin/ee/admin/containers/Roles/CreatePage/ConditionsModal/ActionRow/index.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Text, Padded, Flex } from '@buffetjs/core';
+import { useIntl } from 'react-intl';
+
+import ConditionsSelect from '../ConditionsSelect';
+import Wrapper from './Wrapper';
+
+const ActionRow = ({
+  arrayOfOptionsGroupedByCategory,
+  isGrey,
+  label,
+  name,
+  onCategoryChange,
+  onChange,
+  value,
+}) => {
+  const { formatMessage } = useIntl();
+
+  return (
+    <Wrapper isGrey={isGrey}>
+      <Padded style={{ width: 200 }} top left right bottom size="sm">
+        <Flex>
+          <Text
+            lineHeight="19px"
+            color="grey"
+            fontSize="xs"
+            fontWeight="bold"
+            textTransform="uppercase"
+          >
+            {formatMessage({
+              id: 'Settings.permissions.conditions.can',
+            })}
+            &nbsp;
+          </Text>
+          <Text
+            title={label}
+            lineHeight="19px"
+            fontWeight="bold"
+            fontSize="xs"
+            textTransform="uppercase"
+            color="mediumBlue"
+            style={{ maxWidth: '60%' }}
+            ellipsis
+          >
+            {label}
+          </Text>
+          <Text
+            lineHeight="19px"
+            color="grey"
+            fontSize="xs"
+            fontWeight="bold"
+            textTransform="uppercase"
+          >
+            &nbsp;
+            {formatMessage({
+              id: 'Settings.permissions.conditions.when',
+            })}
+          </Text>
+        </Flex>
+      </Padded>
+      <ConditionsSelect
+        arrayOfOptionsGroupedByCategory={arrayOfOptionsGroupedByCategory}
```

Comment:
ℹ️ I'm feeling that this props tightly coupled UI components (rendered by the ConditionsSelect) with "business" things. I think it's okay for now, but maybe it's something to keep in mind for later (next version and design system), WYT?

Reply:
We will improve the components with the DS, right now I didn't want to change the DOM, basically, when we will update the UI sucn props will be removed, however the business logic will be kept.